addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1191,22 +1192,42 @@ def _send_transient_message(self, partner_to, content):
         })
 
     def execute_command_help(self, **kwargs):
-        partner = self.env.user.partner_id
         if self.channel_type == 'channel':
-            msg = _("You are in channel <b>#%s</b>.", html_escape(self.name))
+            msg = html_escape(_("You are in channel %(bold_start)s#%(channel_name)s%(bold_end)s.")) % {
+                "bold_start": Markup("<b>"),
+                "bold_end": Markup("</b>"),
+                "channel_name": self.name,
+            }
         else:
             all_channel_members = self.env['discuss.channel.member'].with_context(active_test=False)
-            channel_members = all_channel_members.search([('partner_id', '!=', partner.id), ('channel_id', '=', self.id)])
-            msg = _("You are in a private conversation with <b>@%s</b>.", _(" @").join(html_escape(member.partner_id.name or member.guest_id.name) for member in channel_members) if channel_members else _("Anonymous"))
+            channel_members = all_channel_members.search([("is_self", "=", False), ("channel_id", "=", self.id)])
+            if channel_members:
+                member_names = Markup(
+                    format_list(
+                        [f"<b>@%(member_{member.id})s</b>" for member in channel_members],
+                        locale=babel_locale_parse(get_lang(self.env).code),
+                    )
+                ) % {
+                    f"member_{member.id}": member.partner_id.name or member.guest_id.name for member in channel_members
+                }
+                msg = html_escape(_("You are in a private conversation with %(member_names)s.")) % {
+                    "member_names": member_names,
+                }
+            else:
+                msg = _("You are alone in a private conversation.")
         msg += self._execute_command_help_message_extra()
 
```

Comment:
this empty line is useless and confusing, I think it was better removed while we are at it